app/src/main/java/com/weatherxm/ui/devicesrewards/DeviceRewardsAdapter.kt

```diff
@@ -180,6 +191,15 @@ class DeviceRewardsDiffCallback : DiffUtil.ItemCallback<DeviceTotalRewards>() {
         return oldItem.id == newItem.id &&
             oldItem.name == newItem.name &&
             oldItem.total == newItem.total &&
-            oldItem.details == newItem.details
+            oldItem.details == newItem.details &&
+            oldItem.details.total == newItem.details.total &&
+            oldItem.details.mode == newItem.details.mode &&
+            oldItem.details.status == newItem.details.status &&
+            oldItem.details.totals.size == newItem.details.totals.size &&
+            oldItem.details.boosts.size == newItem.details.boosts.size &&
+            oldItem.details.datesChartTooltip.size == newItem.details.datesChartTooltip.size &&
+            oldItem.details.baseChartData == newItem.details.baseChartData &&
+            oldItem.details.betaChartData == newItem.details.betaChartData &&
+            oldItem.details.otherChartData == newItem.details.otherChartData
```

Comment:
Since we have the first check (`oldItem.details == newItem.details`), are the rest required?